src/datasources/erc721.ts

```diff
@@ -40,34 +41,39 @@ import {
 
 export function handleTransfer(event: TransferEvent): void {
 	let contract = fetchERC721(event.address)
-	if (contract != null) {
-		let token = fetchERC721Token(contract, event.params.tokenId)
-		let from  = fetchAccount(event.params.from)
-		let to    = fetchAccount(event.params.to)
-
-		token.owner    = to.id
-		token.approval = fetchAccount(Address.zero()).id // implicit approval reset on transfer
-		token.save()
-
-		let ev         = new ERC721Transfer(events.id(event))
-		ev.emitter     = contract.id
-		ev.transaction = transactions.log(event).id
-		ev.timestamp   = event.block.timestamp
-		ev.contract    = contract.id
-		ev.token       = token.id
-		ev.from        = from.id
-		ev.to          = to.id
-		ev.save()
-	}
+	if (contract == null) return
+
+	let token = fetchERC721Token(contract, event.params.tokenId)
+	let from  = fetchAccount(event.params.from)
+	let to    = fetchAccount(event.params.to)
+
+	token.owner    = to.id
+	token.approval = fetchAccount(Address.zero()).id // implicit approval reset on transfer
+	token.save()
+
+	let ev         = new ERC721Transfer(events.id(event))
+	ev.emitter     = contract.id
+	ev.transaction = transactions.log(event).id
+	ev.timestamp   = event.block.timestamp
+	ev.contract    = contract.id
+	ev.token       = token.id
+	ev.from        = from.id
+	ev.to          = to.id
+	ev.save()
 }
 
 export function handleConsecutiveTransfer(event: ConsecutiveTransfer): void {
 	let contract = fetchERC721(event.address)
-	if (contract != null) {
-		let from  = fetchAccount(event.params.fromAddress)
-		let to    = fetchAccount(event.params.toAddress)
+	if (contract == null) return
+
+	let from        = fetchAccount(event.params.fromAddress)
+	let to          = fetchAccount(event.params.toAddress)
+	let fromTokenId = event.params.fromTokenId.toU64()
+	let toTokenId   = event.params.toTokenId.toU64()
```

Comment:
These are `uint256` in IERC2309. Isn't a way of DoSing the subgraph to emit events with a number higher than `uint256` so the parsing fails here?

Reply:
Possibly. That is interresting.

The u64 parsing was performed for the for loop, but we should indeed use BigInt here.